src/models/settings.js

```diff
@@ -0,0 +1,66 @@
+const mongoose = require("mongoose");
+const mongodbErrorHandler = require("mongoose-mongodb-errors");
+const findOrCreate = require("mongoose-findorcreate");
+
+// Subschema for the Facebook authentication provider credentials.
+const FacebookAuthProviderCredentialsSchema = new mongoose.Schema({});
+
+// Subschema for the Twitter authentication provider credentials.
+const TwitterAuthProviderCredentialsSchema = new mongoose.Schema({});
+
+// Subschema for the Google authentication provider credentials.
+const GoogleAuthProviderCredentialsSchema = new mongoose.Schema({});
+
+// Subschema for the GitHub authentication provider credentials.
+const GitHubAuthProviderCredentialsSchema = new mongoose.Schema({});
+
+// Settings model
+const settingsSchema = new mongoose.Schema({
+  facebookAuthProvider: {
+    appId: {
+      type: String,
+    },
```

Comment:
This is looking good but you did not implement field types? The essence of field types being returned is so that the appropriate UI element will be generated for a given setting. E.g text boxes for string fields or radio buttons for boolean etc

Reply:
Right. Thanks for catching that.